### ISSUE TITLE
Update to Heroku not working?

### ISSUE THREAD

**Issue by johnleeming** [cantino/huginn#985]
Possibly a newbie issue here.  I've set up a Huginn app on Heroku and it's working happily, but I've not been able to update it.  I've created a local clone and then followed the instructions in the documentation.  It happily brings down changes to my local machine but when I push to Heroku it says "everything upto date" and changes nothing.

Sample output from terminal:

john@john­VirtualBox:~/jgl­huginn$ git fetch origin
remote: Counting objects: 37, done.
remote: Total 37 (delta 32), reused 32 (delta 32), pack­reused 5
Unpacking objects: 100% (37/37), done.
From https://github.com/cantino/huginn
   c358b4f..9367ff8  master  ­> origin/master
john@john­VirtualBox:~/jgl­huginn$ git merge origin/master
Updating c358b4f..9367ff8
Fast­forward
 app/models/agents/rss_agent.rb    |   2 +­
 spec/data_fixtures/onethingwell.atom | 221 +++++++++++++++++++++++++++++++++++
 spec/models/agents/rss_agent_spec.rb |  22 +++­
 3 files changed, 240 insertions(+), 5 deletions(­)
 create mode 100644 spec/data_fixtures/onethingwell.atom
john@john­VirtualBox:~/jgl­huginn$ git push ­f heroku master
Everything up­to­date
john@john­VirtualBox:~/jgl­huginn$ heroku maintenance:on
Enabling maintenance mode for jgl­huginn... done
john@john­VirtualBox:~/jgl­huginn$ git push ­f heroku master
Everything up­to­date
john@john­VirtualBox:~/jgl­huginn$ heroku maintenance:off
Disabling maintenance mode for jgl­huginn... done

**Comment by cantino** [comment]
`git push heroku master` should work.  Or try `git push heroku master:master`

What does `git status` show?

**Comment by johnleeming** [comment]
gt push heroku master:master also just say everything uptodaye.

Git status gives:
Not currently on any branch.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
    modified:   app/views/home/_signed_in_index.html.erb
Untracked files:
  (use "git add <file>..." to include in what will be committed)
    app/views/home/_signed_in_index.html.erb~
no changes added to commit (use "git add" and/or "git commit -a")

The change to _signed_in_index.html.erb is an edit to the text I made to check whether anything was having an effect.

Thanks very much for your help

**Comment by johnleeming** [comment]
So now I've managed to assign branches but get the following:

Counting objects: 12819, done.
Compressing objects: 100% (3752/3752), done.
Writing objects: 100% (12819/12819), 2.51 MiB | 872.00 KiB/s, done.
Total 12819 (delta 9565), reused 12123 (delta 8918)
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Fetching custom git buildpack... done
remote: -----> Multipack app detected
remote: =====> Downloading Buildpack: https://github.com/cantino/heroku-selectable-procfile.git
remote: =====> Detected Framework: Selectable Procfile
remote: -----> Using deployment/heroku/Procfile.heroku as Procfile
remote: =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-ruby.git
remote: =====> Detected Framework: Ruby
remote:  !
remote:  !     Your Gemfile.lock contains merge conflicts.
remote:  !     Run `git checkout HEAD -- Gemfile.lock` first to get a clean lock.
remote:  !
remote: /tmp/d20150823-239-nb7mxi/bundler-1.9.7/gems/bundler-1.9.7/lib/bundler/lockfile_parser.rb:36:in `initialize': Your Gemfile.lock contains merge conflicts. (Bundler::LockfileError)
remote: Run `git checkout HEAD -- Gemfile.lock` first to get a clean lock.
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:127:in `new'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:127:in `block in parse_gemfile_lock'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:17:in `block in instrument'
remote: 	from /app/vendor/ruby-2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:16:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:84:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:125:in `parse_gemfile_lock'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:109:in `lockfile_parser'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:72:in `specs'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:57:in `block in gem_version'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:17:in `block in instrument'
remote: 	from /app/vendor/ruby-2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:16:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:84:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/helpers/bundler_wrapper.rb:56:in `gem_version'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/rails5.rb:9:in `block in use?'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:17:in `block in instrument'
remote: 	from /app/vendor/ruby-2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:16:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/base.rb:48:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/rails5.rb:8:in `use?
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack.rb:17:in `block (2 levels) in detect'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack.rb:16:in `each'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack.rb:16:in `detect'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack.rb:16:in `block in detect'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:17:in `block in instrument'
remote: 	from /app/vendor/ruby-2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:16:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack.rb:13:in `detect'
remote: 	from /tmp/buildpackQtkoZ/bin/compile:12:in `block in <main>'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:35:in `call'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:35:in `block in trace'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:17:in `block in instrument'
remote: 	from /app/vendor/ruby-2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:16:in `instrument'
remote: 	from /tmp/buildpackQtkoZ/lib/language_pack/instrument.rb:35:in `trace'
remote: 	from /tmp/buildpackQtkoZ/bin/compile:11:in `<main>'
remote: 
remote:  !     Push rejected, failed to compile Multipack app
remote: 
remote: Verifying deploy....
remote: 
remote: !	Push rejected to jgl-huginn.
remote: 
To https://git.heroku.com/jgl-huginn.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/jgl-huginn.git'
john@john-VirtualBox:~/jgl-huginn$ 
 
I've edited Gemfile.lock to remove any apparent conflicts and git checkout HEAD -- Gemfile.lock but still get the same errors

**Comment by cantino** [comment]
Your git status makes me think that you have uncommitted files or merge conflicts.  `git status` shouldn't show and untracked or modified files; those should be committed and then pushed.  What does `git remote -v` say?  You could paste the Gemfile.lock into a gist if you'd like us to take a look at it.  It sounds like you may have committed merge conflicts into the lock.  If that happened, you could do `git checkout origin/master -- Gemfile.lock` to get the default one, then run `bundle` again.

**Comment by johnleeming** [comment]
I've deleted Gemfile and Gemfile.lock and replaced them with the default one form your master branch.  I've also cleared out the untracked files and my local edits.  So Git status is clear and git remote -v shows the correct addresses for heroku and origin, bundle runs without error but I still get the same error on git push.
Terminal output is here https://drive.google.com/open?id=1_Pt4XncBD012AnwG9iltIw96uUmzR9-YwiP8-W-jtOE
gemfile.lock here: https://drive.google.com/open?id=0B3CP4x1WDC_5R2RnWkJQZ1lQbTg

My next alternative is to deploy a new version of Huginn to Heroku and try copying across my database.

In the meantime the app is working beautifully, I just want keep up with your improvements...
